src/Parallel/ResourceInfo.hpp

```diff
@@ -186,4 +205,687 @@ struct SingletonPack<tmpl::list<ParallelComponents...>> {
  private:
   tuples::tagged_tuple_from_typelist<local_tags> procs_{};
 };
+
+namespace detail {
+template <typename Component>
+using component_has_singleton_info_tag =
+    tmpl::list_contains<typename Component::initialization_tags,
+                        Parallel::Tags::SingletonInfo<Component>>;
+
+template <typename Metavariables>
+using singleton_components =
+    tmpl::filter<typename Metavariables::component_list,
+                 Parallel::is_singleton<tmpl::_1>>;
+
+template <typename Component>
+using contains_avoid_global_proc_0 =
+    tmpl::list_contains<typename Component::initialization_tags,
+                        Parallel::Tags::AvoidGlobalProc0>;
+
+template <typename Metavariables>
+constexpr bool avoid_global_proc_0 =
+    tmpl::size<tmpl::filter<
+        typename Metavariables::component_list,
+        tmpl::bind<contains_avoid_global_proc_0, tmpl::_1>>>::value > 0;
+
+template <typename SingletonList>
+using singletons_with_singleton_info =
+    tmpl::filter<SingletonList,
+                 tmpl::bind<component_has_singleton_info_tag, tmpl::_1>>;
+
+template <typename Metavariables>
+constexpr bool using_resource_info =
+    avoid_global_proc_0<Metavariables> or
+    tmpl::size<singletons_with_singleton_info<
+        singleton_components<Metavariables>>>::value > 0;
+
+}  // namespace detail
+
+/*!
+ * \ingroup ParallelGroup
+ * \brief Holds resource info for all singletons and for avoiding placing array
+ * elements/singletons on the global proc 0.
+ *
+ * \details This can be used for placing all singletons in an executable. To
+ * have a singleton specify resource information, add the
+ * `Parallel::Tags::SingletonInfo` tag to the `initialization_tags` type alias
+ * in the singleton. To specify whether to avoid placing array elements and
+ * singletons on the global proc 0, add the `Parallel::Tags::AvoidGlobalProc0`
+ * tag to the `initialization_tags` of any parallel component in your
+ * executable.
```

Comment:
At the beginning of the sentence I say `To specify whether to avoid placing array elements and singletons on the global proc 0, ...` so I'm not indicating that (node)groups won't be placed on global proc 0. I just check the initialization tags of all components for the AvoidGlobalProc0 tag, but the tag will only have an effect on the placement of singletons and arrays.

Should I reword this to make it clearer?

Reply:
Hmm, agreed. Maybe `of the... components` (last pluralize). But up to you, I agree you do give the restriction at the beginning so if someone encounters and issue they can reread and it should clarify :)